src/ray/object_manager/object_buffer_pool.h

```diff
@@ -0,0 +1,193 @@
+  /// \return The buffer length of the chunk at chunk_index.
+  uint64_t GetBufferLength(uint64_t chunk_index, uint64_t data_size);
+
+  /// Returns a chunk of an object at the given chunk_index. The object chunk serves
+  /// as the data that is to be written to a connection as part of sending an object to
+  /// a remote node.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param data_size The sum of the object size and metadata size.
+  /// \param metadata_size The size of the metadata.
+  /// \param chunk_index The index of the chunk.
+  /// \return A pair consisting of a ChunkInfo struct and status of invoking this method.
+  std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> GetChunk(const ObjectID &object_id, uint64_t data_size,
+                            uint64_t metadata_size, uint64_t chunk_index);
+
+  /// When a chunk is done being used as part of a get, this method is invoked
+  /// to indicate that the buffer associated with a chunk is no longer needed.
+  ///
+  /// \param object_id The object_id of the buffer to release.
+  /// \param chunk_index The index of the chunk.
+  /// \return The status of invoking this method.
+  ray::Status ReleaseGetChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+  /// Returns a chunk of an empty object at the given chunk_index. The object chunk
+  /// serves as the data that is to be written to by a connection receiving an object
+  /// from a remote node. Only one chunk can be referenced at a time.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param data_size The sum of the object size and metadata size.
+  /// \param metadata_size The size of the metadata.
+  /// \param chunk_index The index of the chunk.
+  /// \return A ChunkInfo struct.
+  std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> CreateChunk(const ObjectID &object_id, uint64_t data_size,
+                               uint64_t metadata_size, uint64_t chunk_index);
+
+  /// Abort the create operation associated with a chunk at chunk_index.
+  /// This should not be called if SealChunk is called.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param chunk_index The index of the chunk.
+  /// \return The status of invoking this method.
+  ray::Status AbortCreateChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+  /// Seal the object associated with a create operation. This is invoked whenever
+  /// a chunk is successfully written to. This is not called if AbortCreateChunk
+  /// is called.
+  ///
+  /// \param object_id The ObjectID.
+  /// \param chunk_index The index of the chunk.
+  /// \return The status of invoking this method.
+  ray::Status SealChunk(const ObjectID &object_id, uint64_t chunk_index);
+
+ private:
+
+  /// Abort the create operation associated with an object. This destroys the buffer
+  /// state, including create operations in progress for all chunks of the object.
+  ///
+  /// \param object_id The ObjectID.
+  /// \return The status of invoking this method.
```

Comment:
For methods that return `ray::Status`, we should be clear about when/how they might fail. For this method and some others on this class, it looks like the only time they should fail is if there's a bug, which we can catch with `RAY_CHECK` calls. I would make any of these methods return `void` instead.